File: result/metagpt/claude/humaneval/132.c
#include <stdbool.h>
#include <string.h>

bool is_nested(const char* str) {
    int count = 0, maxcount = 0;
    int length = strlen(str);
    
    for (int i = 0; i < length; i++) {
        if (str[i] == '[') count += 1;
        if (str[i] == ']') count -= 1;
        if (count < 0) count = 0;
        if (count > maxcount) maxcount = count;
        if (count <= maxcount - 2) return true;
    }
    
    return false;
}

File: result/metagpt/gpt4o/humaneval/132.rs
fn is_nested(str: &str) -> bool {
    let mut count = 0;
    let mut maxcount = 0;

    for ch in str.chars() {
        if ch == '[' {
            count += 1;
        } else if ch == ']' {
            count -= 1;
        }

        if count < 0 {
            count = 0;
        }

        if count > maxcount {
            maxcount = count;
        }

        if count <= maxcount - 2 {
            return true;
        }
    }

    false
}

fn main() {
    let test_str = "[[[]]]";
    let result = is_nested(test_str);
    println!("Is the string nested? {}", result);
}